### ISSUE TITLE
Event without arguments outputs Exception (compilation error)

### ISSUE THREAD

**Issue by gsmachado** [neow3j/neow3j#365]
WHen I try to compile this contract:

```
package io.neow3j.examples.contractdev.contracts;

import io.neow3j.devpack.annotations.DisplayName;
import io.neow3j.devpack.annotations.ManifestExtra;
import io.neow3j.devpack.events.Event;

@ManifestExtra(key = "name", value = "SingleArgumentContract")
@ManifestExtra(key = "author", value = "AxLabs")
public class SingleArgumentContract {

    @DisplayName("onEvent")
    static Event onEvent;

    public static boolean anything() throws Exception {
        onEvent.notify();
        return true;
    }
}
```

The output is:

```
Exception in thread "main" java.lang.IndexOutOfBoundsException
        at org.objectweb.asm.tree.InsnList.get(InsnList.java:94)
        at io.neow3j.compiler.NeoMethod.convert(NeoMethod.java:474)
        at io.neow3j.compiler.converters.MethodsConverter.handleUncachedMethodCall(MethodsConverter.java:183)
        at io.neow3j.compiler.converters.MethodsConverter.handleMethodCall(MethodsConverter.java:157)
        at io.neow3j.compiler.converters.MethodsConverter.handleInvoke(MethodsConverter.java:138)
        at io.neow3j.compiler.converters.MethodsConverter.convert(MethodsConverter.java:73)
        at io.neow3j.compiler.Compiler.handleInsn(Compiler.java:388)
        at io.neow3j.compiler.NeoMethod.convert(NeoMethod.java:476)
        at io.neow3j.compiler.Compiler.compileClass(Compiler.java:242)
        at io.neow3j.compiler.Compiler.compileClass(Compiler.java:214)
        at io.neow3j.examples.contractdev.CompileAndDeploy.main(CompileAndDeploy.java:41)
```

It seems that the problem is the `Event`.

So... if I change to `Event2Args` it works:

```
package io.neow3j.examples.contractdev.contracts;

import io.neow3j.devpack.annotations.DisplayName;
import io.neow3j.devpack.annotations.ManifestExtra;
import io.neow3j.devpack.events.Event;

@ManifestExtra(key = "name", value = "SingleArgumentContract")
@ManifestExtra(key = "author", value = "AxLabs")
public class SingleArgumentContract {

    @DisplayName("onEvent")
    static Event2Args<String, String> onEvent;

    public static boolean anything() throws Exception {
        onEvent.notify("test", "test");
        return true;
    }
}
```

But, if I try to compile the following (only using `onEvent.notify()` in a `Event2Args` type), it miserably fails:

```
package io.neow3j.examples.contractdev.contracts;

import io.neow3j.devpack.annotations.DisplayName;
import io.neow3j.devpack.annotations.ManifestExtra;
import io.neow3j.devpack.events.Event;

@ManifestExtra(key = "name", value = "SingleArgumentContract")
@ManifestExtra(key = "author", value = "AxLabs")
public class SingleArgumentContract {

    @DisplayName("onEvent")
    static Event2Args<String, String> onEvent;

    public static boolean anything() throws Exception {
        onEvent.notify();
        return true;
    }
}
```

In my opinion, the compiler should either not support the "Event" without parameters, or provide a better error message, or provide a way that the developer can use `Event`.

**Comment by csmuller** [comment]
😲  it's the `Object.notify()` method you used there. I only see that name collision now 😒  That method is not intended to be used on events (or on anything in the devpack). But it looks like we need to find a different naming for the `notify(...)` methods to prevent this misunderstanding.

**Comment by gsmachado** [comment]
@csmuller oh, I see the problem now.

What about if we rename to `notifyEvent()`? Anything else in mind?

**Comment by csmuller** [comment]
I'd rather move away from `notify` completely and change it to `send` or `trigger. What do you think?

**Comment by gsmachado** [comment]
> I'd rather move away from `notify` completely and change it to `send` or `trigger`. What do you think?

Hmmm... agreed.

I think `.trigger()` is more appropriate than `.send()`. But I'm not strongly opinionated on this. It can be either one. 😄

**Comment by csmuller** [comment]
😃 Good, then we'll go with `trigger()` 👍

**Comment by csmuller** [comment]
- [ ] Rename the `notify()` method of all devpack events to `trigger()`.
- [ ] Add an event class that supports no parameters (e.g. `Event0Args`)
- [ ] Possibly redesign the usage of events. E.g., make it mandatory to instantiate the event (like it is normally required for Java object variables before use.), remove the `Event` interface and replace it with an annotation to prohibit the usage of the interface in smart contract code.

**Comment by gsmachado** [comment]
@csmuller why did you write:

> Remove `Runtime.notify()` method.

?

Another thing: maybe instead of `.trigger()` (as we agreed beforehand), we should consider renaming to something else.

The reason is: there are some methods in the `Runtime` class already that maybe could confuse with the event trigger:

```
    @Syscall(SYSTEM_RUNTIME_GETTRIGGER)
    public static native byte getTrigger();
```

So, I suggest using 1) `event.log()` or 2) `event.fire()` or 3) `event.send()`. OR, if you think it's feasible, we could use `@Override` in all `Event` methods, in order to override the `Object.notify()`.

What do you think?

Let me know. I'm willing to implement and close this issue before we release it.

**Comment by gsmachado** [comment]
oh, ok, I think `@Override` is not a possibility:

![image](https://user-images.githubusercontent.com/2124276/111711933-fe48ad80-884c-11eb-9c24-333174d3ac6a.png)

**Comment by csmuller** [comment]
I prefer `event.fire()` 👍 
You can go ahead and apply that change if you want to.

**Comment by csmuller** [comment]
Solved in #478.
